test/core.jl

```diff
@@ -539,6 +539,19 @@ end
 @test a[2](10) == 12
 @test a[3](10) == 13
 
+# issue #22032
+let a = []
+    for f() in 1:3
+        push!(a, f())
+    end
+    @test a == [1,2,3]
+end
```

Comment:
Maybe also this test:

```jl
let a = []
    for f() in 1:3
        push!(a, f)
    end
    @test [f() for f in a] == [1,2,3]
end
```

Just to make sure that the closure capture is working as it should.